docs/dev_guide/contents/extending_fido.rst

```diff
@@ -0,0 +1,501 @@
+Please note that it can be defined in a similar way for full clients too.
+
+An Example of ``register_values()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    @classmethod
+    def register_values(cls):
+
+        from sunpy.net import attrs
+        adict = {
+        attrs.Instrument: [("LASCO", "Large Angle and Spectrometric Coronagraph")],
+        attrs.Source: [('SOHO', 'Solar and Heliospheric Observatory')],
+        attrs.Provider: [('SDAC', 'Solar Data Analysis Center')],
+        attrs.Detector: [('C1', 'Coronograph 1'),
+                         ('C2', 'Coronograph 2'),
+                         ('C3', 'Coronograph 3')]
+        }
+
+        return adict
+
+Writing a Search Method
+-----------------------
+
+The ``search()`` method has the job of taking a set of user queries and returning an instance of `.QueryResponseTable` containing the results.
+
+The general flow of a ``search()`` method is:
+
+* Call your instance of an `.AttrWalker` to convert the input into a form expected by your API.
+* Make as many requests to your API as needed to furfill the query. (Generally one per element of the outer `.AttrOr`).
+* Process the response from your API into an instance of `.QueryResponseTable`.
+
+To process the query with the `.AttrWalker`, call the :meth:`.AttrWalker.create` method::
+
+  def search(self, query):
+    queries = walker.create(query)
+
+Assuming the walker is the one we defined above, queries would be a list of dicts with the attrs processed into query parameters for the API URL.
+
+.. note::
+
+   If you want your search method to be able to be called independently of Fido, then you should accept a variable number of positional arguments (``*args``) and they should have the AND operator applied to them.
+   This looks like::
+
+     def search(self, *args):
+         query = attr.and_(args)
+         queries = walker.create(query)
+
+
+Once the walker has processed the query into a form designed to be passed to your API, your ``search()`` method then needs to iterate over these parameters, make the requests and process the results in to a table.
+
+In the following example we pretend our client has a method ``_make_search(query_parameters)`` which takes the query parameters and makes a request to our API.
+We also pretend that the response is a json object in the form of a Python dictionary, which we want to put into the table.
+
+.. code-block:: python
+
+  def search(self, query):
+      queries = walker.create(query)
+
+      results = []
+      for query_parameters in queries:
+          results.append(self._make_search(query_parameters))
+
+      return QueryResponseTable(results, client=self)
+
+
+In reality, you probably want to post-process the results from your API before you put them in the table, they should be human readable first, with spaces and capitalisation as appropriate.
+
+
+The ``_can_handle_query`` method
```

Comment:
it should be noted here that this is also required for the simple "scraper"-type client too, not just the "full"client

Reply:
It isn't anymore. One is autogenerated from the contents of `register_values` I think.

Reply:
ah yas sorry my bad